pkg/awsutils/awsutils.go

```diff
@@ -455,7 +443,10 @@ func (cache *EC2InstanceMetadataCache) refreshSGIDs(mac string) error {
 			awsAPILatency.WithLabelValues("ModifyNetworkInterfaceAttribute", fmt.Sprint(err != nil)).Observe(msSince(start))
 			if err != nil {
 				awsAPIErrInc("ModifyNetworkInterfaceAttribute", err)
-				return errors.Wrap(err, "refreshSGIDs: unable to update the ENI's SG")
+				//No need to return error here since retry will happen in 30seconds and also
+				//If update failed due to stale ENI then returning error will prevent updating SG 
```

Comment:
There are two aspects to metadata not being in sync 1) transient 2) permanent (let's say some bug in the underlying metadata provider) 

This change handles # 1 well. We should have a way to bubble # 2 to the user so they can raise an issue on the dependency if # 2 ever happens and work towards remedy

Reply:
Yeah makes sense, I will make the changes and update the PR.